src/agents/memory/async_redis_checkpointer.py

```diff
@@ -528,3 +529,13 @@ async def alist_llm_usage_records(self, cluster_id: str, ttl: int) -> list[dict]
                 latest_keys.append(key)
         records = await self.conn.mget(latest_keys)
         return [json.loads(record) for record in records if record]
+
+
+def get_async_redis_saver() -> AsyncRedisSaver:
+    """
+    Returns an instance of AsyncRedisSaver with a async redis connection as defined in the config files.
+    """
+    connection = get_redis_connection().__next__().connection
```

Comment:
Is it a good idea to use a private class method `__next__` like this?

Reply:
I just noticed that I no longer need a generator so I will just refactor that func. Makes everything easier.